src/explore-education-statistics-common/src/components/Tabs.tsx

```diff
@@ -67,7 +69,9 @@ const Tabs = ({ children, id, onToggle }: Props) => {
       <ul className="govuk-tabs__list" role="tablist">
         {sections.map(({ props }, index) => {
           const sectionId = props.id || `${id}-${index + 1}`;
-
+          if (openId && openId === sectionId && selectedTabIndex !== index) {
+            setSelectedTab(index);
```

Comment:
It's dangerous to be setting state within the component's render. Typically you only should be doing this in `useEffect` or callback functions.

I'm also getting the impression that setting `openId` will lock the tabs to that specific section, which seems like it would probably be incorrect behaviour as you won't be able to open another tab.